flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -308,6 +327,10 @@ private <T> DataStreamSink<T> chainIcebergOperators() {
         }
       }
 
+      Map<String, String> mergedProps = Maps.newHashMap(table.properties());
+      mergedProps.putAll(props);
+      this.props = mergedProps;
```

Comment:
it is a little weird to me to update the `this.props`. We can just pass the `mergedProps` to the methods where merged props are needed.